shm/Island.hpp

```diff
@@ -11,20 +11,21 @@ class Island {
 
     class Coordinates {
     public:
+        static constexpr size_t distance(const Coordinates& lhs, const Coordinates& rhs) {
+            auto pairX = std::minmax(lhs.positionX_, rhs.positionX_);
+            auto pairY = std::minmax(lhs.positionY_, rhs.positionY_);
```

Comment:
std::minmax -> it is really necessary ? In my opinion, equation below will be fine without minmax - we can just square a negative number .

Reply:
size_t dont have negative number.